src/js/components/FormField/FormField.js

```diff
@@ -380,6 +394,14 @@ const FormField = forwardRef(
           if (contextOnBlur) contextOnBlur(event);
           if (onBlur) onBlur(event);
         }}
+        onChange={event => {
+          const debouncedFn =  debounce(() => {
+            if (contextOnChange) contextOnChange(event);
+            if (onChange) onChange(event);
```

Comment:
makes sense, will do !

Reply:
We will still end up in a nested ternary inside the function, to call it unless I am missing something here.

Reply:
I also think we have to avoid putting `onChange` inside the debounce to avoid any backwards compatibility issues.

What about?:
```
onChange={
  (contextOnChange || onChange)
    ? event => {
      event.persist();
      if (onChange) onChange(event);
      if (contextOnChange) {
        const debouncedFn =  debounce(() => {
          contextOnChange(event);
        }, 500);
        debouncedFn();
      }
    }
    : undefined
}
```